File: go/internal/slides/13_callingconv/main.go
package main

import "fmt"

// Add three numbers
//
//go:noinline
func Add(x, y int) int {
	return x + y
}

// Entrypoint to illustrate calling convention
func main() {
	x := 1
	y := 2

	r := Add(x, y)

	fmt.Println(r)
}

File: go/internal/slides/13_callingconv/main.c
#include <stdio.h>

__attribute__ ((noinline)) int add(int x, int y) {
    return x + y;
}

int main() {
    int x = 1;
    int y = 2;

    int r = add(x, y);

    printf("%i\n", r);
}

